draft-ietf-quic-transport.md

```diff
@@ -1607,22 +1606,30 @@ NOT subsequently advertise a smaller offset.  A sender may receive WINDOW_UPDATE
 frames out of order; a sender MUST therefore ignore any reductions in flow
 control credit.
 
+A receiver MUST close the connection with a
+QUIC_FLOW_CONTROL_RECEIVED_TOO_MUCH_DATA error ({{error-handling}}) if the
+peer violates the advertised stream or connection flow control windows.
+
 A sender MUST send BLOCKED frames to indicate it has data to write but is
 blocked by lack of connection or stream flow control credit.  BLOCKED frames are
 expected to be sent infrequently in common cases, but they are considered useful
 for debugging and monitoring purposes.
 
 A receiver advertises credit for a stream by sending a WINDOW_UPDATE frame with
-the StreamID set appropriately.  A receiver may simply use the current received
-offset to determine the flow control offset to be advertised.
+the StreamID set appropriately. A receiver may use the current offset of data
+consumed to determine the flow control offset to be advertised.
+A receiver MAY send copies of a WINDOW_UPDATE frame in multiple packets in order
+to make sure that the sender receives it even if one of the packets is lost. A
+sender MUST ignore duplicate WINDOW_UPDATE frames.
```

Comment:
Yeah, I think this is a nice optimization. +1 to Martin's point about ignoring "old" WINDOW_UPDATE frames.

The PR doesn't quite capture the point that a sender must ensure that a WINDOW_UPDATE frame is delivered to the receiver, which the draft currently guarantees by making the frame reliable. What you want is that the most recent value be the one that is transmitted. 

We could define retransmission of control frames as allowing newer control information. This would apply to WINDOW_UPDATE, PING, RST_STREAM, CONNECTION_CLOSE frames, but the only place it would be really useful is in WINDOW_UPDATE. WDYT?

Beyond that, we can allow senders to send multiple WINDOW_UPDATE frames if they want, since, as you suggest in your PR, the receiver MUST ignore "old" ones.